pkg/proxy/proxy.go

```diff
@@ -31,21 +31,19 @@ var (
 	ErrFileToStringEmpty error = errors.New("file is empty")
 )
 
-type loadTestSpecCreator func(*http.Request, backends.Config, *zap.Logger) (apisLoadTestV1.LoadTestSpec, error)
-
 // Proxy handler
 type Proxy struct {
-	config              Config
-	kubeClient          *kube.Client
-	httpToSpecConverter loadTestSpecCreator
+	maxLoadTestsRun int
+	registry        *backends.Registry
+	kubeClient      *kube.Client
 }
 
 // NewProxy returns new Proxy handlers
-func NewProxy(cfg Config, kubeClient *kube.Client, specCreator loadTestSpecCreator) *Proxy {
+func NewProxy(maxLoadTestsRun int, registry *backends.Registry, kubeClient *kube.Client) *Proxy {
```

Comment:
Why are we expecting a `struct` and not an interface? (same question for the other places where `*backends.Registry` is used) To me this seems to make the code a lot harder to test properly. Maybe use and interface?